backend_new/src/services/email.service.ts

```diff
@@ -0,0 +1,345 @@
+import { Application } from '../dtos/applications/application.dto';
+
+type User = {
+  firstName: string;
+  middleName: string;
+  lastName: string;
+  email: string;
+  language: LanguagesEnum;
+};
+@Injectable()
+export class EmailService {
+  polyglot: Polyglot;
+
+  constructor(
+    private readonly sendGrid: SendGridService,
+    private readonly configService: ConfigService,
+    private readonly translationService: TranslationService,
+    private readonly jurisdictionService: JurisdictionService,
+  ) {
+    this.polyglot = new Polyglot({
+      phrases: {},
+    });
+    const polyglot = this.polyglot;
+    Handlebars.registerHelper(
+      't',
+      function (
+        phrase: string,
+        options?: number | Polyglot.InterpolationOptions,
+      ) {
+        return polyglot.t(phrase, options);
+      },
+    );
+    const parts = this.partials();
+    Handlebars.registerPartial(parts);
+  }
+
+  private template(view: string) {
+    return Handlebars.compile(
+      fs.readFileSync(
+        path.join(path.resolve(__dirname, '..', 'views'), `/${view}.hbs`),
+        'utf8',
+      ),
+    );
+  }
+
+  private partial(view: string) {
+    return fs.readFileSync(
+      path.join(path.resolve(__dirname, '..', 'views'), `/${view}`),
+      'utf8',
+    );
+  }
+
+  private partials() {
+    const partials = {};
+    const dirName = path.resolve(__dirname, '..', 'views/partials');
+
+    fs.readdirSync(dirName).forEach((filename) => {
+      partials[filename.slice(0, -4)] = this.partial('partials/' + filename);
+    });
+
+    const layoutsDirName = path.resolve(__dirname, '..', 'views/layouts');
+
+    fs.readdirSync(layoutsDirName).forEach((filename) => {
+      partials[`layout_${filename.slice(0, -4)}`] = this.partial(
+        'layouts/' + filename,
+      );
+    });
+
+    return partials;
+  }
+
+  private async send(
+    to: string,
+    from: string,
+    subject: string,
+    body: string,
+    retry = 3,
+  ) {
+    await this.sendGrid.send(
+      {
+        to: to,
+        from,
+        subject: subject,
+        html: body,
+      },
+      false,
+      (error) => {
+        if (error instanceof ResponseError) {
+          const { response } = error;
+          const { body: errBody } = response;
+          console.error(
+            `Error sending email to: ${to}! Error body: ${errBody}`,
+          );
+          if (retry > 0) {
+            void this.send(to, from, subject, body, retry - 1);
+          }
+        }
+      },
+    );
+  }
+
+  private async loadTranslations(
+    jurisdiction: Jurisdiction | null,
+    language?: LanguagesEnum,
+  ) {
+    const translations = await this.translationService.getMergedTranslations(
+      jurisdiction?.id,
+      language,
+    );
+    this.polyglot.replace(translations);
+  }
+
+  private async getJurisdiction(
+    jurisdictionIds: IdDTO[] | null,
+  ): Promise<Jurisdiction | null> {
+    // TODO: figure out multiple jurisdictions
```

Comment:
What's the next step for this TODO / issue?

Reply:
I decided to remove the comment since the way it works with my latest change is that if more than one jurisdiction exists we only send the generic translations (null jurisdiction)